### ISSUE TITLE
[P0] Agent design phase consumes excessive tokens due to missing v3 example in prompt

### ISSUE THREAD

**Issue by kpiteira** [kpiteira/ktrdr#253]
## Priority: P0 (Critical)

The agent design phase consumes 50K+ tokens because Claude must guess the v3 strategy format through trial-and-error validation cycles. A detailed v3 example in the prompt would reduce this to ~10K tokens.

## Summary

The agent needed 5+ iterations of `validate_strategy_config` because the prompt lacks a complete v3 strategy example. Each validation cycle adds to cumulative token cost (which is billed). The 50K token limit is reasonable - the agent shouldn't need anywhere near that to design a simple strategy.

## Evidence

From backend logs for operation `op_agent_design_20260119_053108_cee0fc61`:

```
iteration=1: input_tokens=7,151, output=970, tool=validate_strategy_config
iteration=2: input_tokens=7,955, output=616, tool=validate_strategy_config  
iteration=3: input_tokens=8,790, output=650, tool=validate_strategy_config
iteration=4: input_tokens=9,490, output=630, tool=validate_strategy_config
iteration=5: input_tokens=10,509, output=620, tool=validate_strategy_config
iteration=6: FAILED - cumulative tokens exceeded 50K limit
```

**What happened:**
- Claude guessed v3 format → validated → got errors → tried again
- Each retry adds the full conversation history to next call
- Cumulative billing: 7151+7955+8790+9490+10509 = **43,895 input tokens**
- Plus ~3,500 output tokens
- All wasted on trial-and-error

**Validation errors Claude encountered:**
- "Missing fuzzy_sets for indicators: rsi_14"
- "Unknown indicator type 'rsi_14'" (case sensitivity)
- "Orphaned fuzzy_sets detected"

These errors show Claude was guessing the v3 structure.

## Root Cause

The prompt in `ktrdr/agents/prompts.py` lacks a complete, validated v3 strategy example. Claude has to guess:

- That `indicators` is a DICT keyed by ID, not a list
- That `nn_inputs` section is required
- How `fuzzy_sets` link to indicators via the `indicator` field
- Exact field names, nesting, and case sensitivity

## Required Fix

Add a complete v3 example to `SYSTEM_PROMPT_TEMPLATE` in `ktrdr/agents/prompts.py`:

```markdown
## V3 Strategy Format (REQUIRED)

You MUST use this exact structure. Copy this pattern exactly:

```yaml
name: my_strategy
description: Brief description
version: '3.0'
scope: universal

training_data:
  symbols:
    mode: single
    symbol: EURUSD
  timeframes:
    mode: single
    list: [1h]
    base_timeframe: 1h
  history_required: 100

# Indicators are a DICT keyed by unique ID (NOT a list!)
indicators:
  rsi_14:                    # This ID is referenced in fuzzy_sets
    type: RSI                # Case-sensitive! Must match exactly
    period: 14
    source: close

# Fuzzy sets interpret indicator values
fuzzy_sets:
  rsi_14:                    # Must match an indicator ID above
    oversold:
      type: triangular
      parameters: [0, 20, 40]
    neutral:
      type: triangular
      parameters: [30, 50, 70]
    overbought:
      type: triangular
      parameters: [60, 80, 100]
    indicator: rsi_14        # Required: links to indicator

# nn_inputs - REQUIRED section
nn_inputs:
  - fuzzy_set: rsi_14        # References fuzzy_set key
    timeframes: all

model:
  type: mlp
  architecture:
    hidden_layers: [16, 8]
    activation: relu
    output_activation: softmax
    dropout: 0.1
  training:
    learning_rate: 0.001
    batch_size: 32
    epochs: 10

decisions:
  output_format: classification
  confidence_threshold: 0.6

training:
  method: supervised
  labels:
    source: zigzag
    zigzag_threshold: 0.02
```

**Critical v3 rules:**
1. `indicators` is a DICT keyed by ID, NOT a list
2. Each indicator ID MUST have a matching `fuzzy_sets` entry
3. Each fuzzy_set MUST have `indicator: <id>` field linking back
4. `nn_inputs` section is REQUIRED
5. Indicator `type` is CASE-SENSITIVE (use RSI not rsi)
```

## Expected Outcome

With the v3 example in the prompt:
- Claude gets format right on iteration 1 (maybe 2 for complex strategies)
- Token consumption: ~14K instead of 50K+
- **~70% cost reduction per design**

## Files to Modify

- `ktrdr/agents/prompts.py` - Add v3 example to SYSTEM_PROMPT_TEMPLATE

## Acceptance Criteria

- [ ] Complete v3 example added to prompt
- [ ] Agent designs simple strategy in ≤2 iterations
- [ ] Token consumption for simple design < 20K
